<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1002</width>
    <height>738</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Db models creator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="Ui.qrc">
    <normaloff>:/Icons/Anal.png</normaloff>:/Icons/Anal.png</iconset>
  </property>
  <property name="iconSize">
   <size>
    <width>28</width>
    <height>28</height>
   </size>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="2">
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>C++ header</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonModelHeaderCopy">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="Ui.qrc">
            <normaloff>:/Icons/Copy.png</normaloff>:/Icons/Copy.png</iconset>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="plainTextEditModelHeader"/>
      </item>
     </layout>
    </item>
    <item row="1" column="3">
     <layout class="QVBoxLayout" name="verticalLayout_8">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>C++ source</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonModelSourceCopy">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="Ui.qrc">
            <normaloff>:/Icons/Copy.png</normaloff>:/Icons/Copy.png</iconset>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="plainTextEditModelSource"/>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Db script</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonSourcePaste">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="Ui.qrc">
            <normaloff>:/Icons/Paste.png</normaloff>:/Icons/Paste.png</iconset>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="plainTextEditDbScript">
        <property name="plainText">
         <string>CREATE TABLE object_state_log
(
  _id serial NOT NULL,
  _ostate integer NOT NULL,
  old_state integer NOT NULL,
  new_state integer NOT NULL,
  change_time timestamp with time zone NOT NULL,
  CONSTRAINT object_state_log_pkey PRIMARY KEY (_id),
  CONSTRAINT object_state_log__ostate_fkey FOREIGN KEY (_ostate)
      REFERENCES object_state (_id) MATCH SIMPLE
      ON UPDATE CASCADE ON DELETE CASCADE
)
</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QToolButton" name="toolButtonGenerate">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="Ui.qrc">
          <normaloff>:/Icons/Anal.png</normaloff>:/Icons/Anal.png</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolButtonSave">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="Ui.qrc">
          <normaloff>:/Icons/Save.png</normaloff>:/Icons/Save.png</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolButtonPath">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="Ui.qrc">
          <normaloff>:/Icons/Folder.png</normaloff>:/Icons/Folder.png</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap resource="Ui.qrc">:/Icons/SeparatorV.png</pixmap>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolButtonBatch">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="Ui.qrc">
          <normaloff>:/Icons/Script.png</normaloff>:/Icons/Script.png</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="2">
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>C++ header</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonHeaderCopy">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="Ui.qrc">
            <normaloff>:/Icons/Copy.png</normaloff>:/Icons/Copy.png</iconset>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="plainTextEditHeader"/>
      </item>
     </layout>
    </item>
    <item row="0" column="3">
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>C++ source</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="toolButtonSourceCopy">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="Ui.qrc">
            <normaloff>:/Icons/Copy.png</normaloff>:/Icons/Copy.png</iconset>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="plainTextEditSource"/>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_9">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label_9">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>16</height>
           </size>
          </property>
          <property name="text">
           <string>Class names</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_10">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>16</height>
           </size>
          </property>
          <property name="text">
           <string>suffix</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEditPrefix">
          <property name="text">
           <string>Z</string>
          </property>
          <property name="placeholderText">
           <string>Prefix</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="plainTextEditClasses"/>
      </item>
     </layout>
    </item>
    <item row="1" column="1">
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>32</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QToolButton" name="toolButtonSaveModel">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="Ui.qrc">
          <normaloff>:/Icons/Save.png</normaloff>:/Icons/Save.png</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolButtonPathModel">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="Ui.qrc">
          <normaloff>:/Icons/Folder.png</normaloff>:/Icons/Folder.png</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap resource="Ui.qrc">:/Icons/SeparatorV.png</pixmap>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="toolButtonBatchModel">
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="Ui.qrc">
          <normaloff>:/Icons/Script.png</normaloff>:/Icons/Script.png</iconset>
        </property>
        <property name="autoRaise">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionGenerate">
   <property name="icon">
    <iconset resource="Ui.qrc">
     <normaloff>:/Icons/Anal.png</normaloff>:/Icons/Anal.png</iconset>
   </property>
   <property name="text">
    <string>Generate</string>
   </property>
   <property name="toolTip">
    <string>Generate model</string>
   </property>
  </action>
  <action name="actionLong">
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="Ui.qrc">
     <normaloff>:/Icons/64bit.png</normaloff>:/Icons/64bit.png</iconset>
   </property>
   <property name="text">
    <string>64bit</string>
   </property>
   <property name="toolTip">
    <string>Use 64 bit integer</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="Ui.qrc">
     <normaloff>:/Icons/Save.png</normaloff>:/Icons/Save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save table as .cpp, .h</string>
   </property>
  </action>
  <action name="actionSaveModel">
   <property name="icon">
    <iconset resource="Ui.qrc">
     <normaloff>:/Icons/Save.png</normaloff>:/Icons/Save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save table model as .cpp, .h</string>
   </property>
  </action>
  <action name="actionPath">
   <property name="icon">
    <iconset resource="Ui.qrc">
     <normaloff>:/Icons/Folder.png</normaloff>:/Icons/Folder.png</iconset>
   </property>
   <property name="text">
    <string>Browse</string>
   </property>
   <property name="toolTip">
    <string>Browse save path</string>
   </property>
  </action>
  <action name="actionPathModel">
   <property name="icon">
    <iconset resource="Ui.qrc">
     <normaloff>:/Icons/Folder.png</normaloff>:/Icons/Folder.png</iconset>
   </property>
   <property name="text">
    <string>Browse</string>
   </property>
   <property name="toolTip">
    <string>Browse save model path</string>
   </property>
  </action>
  <action name="actionBatch">
   <property name="icon">
    <iconset resource="Ui.qrc">
     <normaloff>:/Icons/Script.png</normaloff>:/Icons/Script.png</iconset>
   </property>
   <property name="text">
    <string>Batch</string>
   </property>
   <property name="toolTip">
    <string>Batch execute</string>
   </property>
  </action>
  <action name="actionBatchModel">
   <property name="icon">
    <iconset resource="Ui.qrc">
     <normaloff>:/Icons/Script.png</normaloff>:/Icons/Script.png</iconset>
   </property>
   <property name="text">
    <string>Batch</string>
   </property>
   <property name="toolTip">
    <string>Batch execute (models)</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>toolButtonSourcePaste</tabstop>
  <tabstop>toolButtonGenerate</tabstop>
  <tabstop>toolButtonSave</tabstop>
  <tabstop>toolButtonHeaderCopy</tabstop>
  <tabstop>toolButtonSourceCopy</tabstop>
  <tabstop>lineEditPrefix</tabstop>
  <tabstop>toolButtonModelHeaderCopy</tabstop>
  <tabstop>toolButtonModelSourceCopy</tabstop>
  <tabstop>plainTextEditDbScript</tabstop>
  <tabstop>plainTextEditHeader</tabstop>
  <tabstop>plainTextEditSource</tabstop>
  <tabstop>plainTextEditClasses</tabstop>
  <tabstop>plainTextEditModelHeader</tabstop>
  <tabstop>plainTextEditModelSource</tabstop>
 </tabstops>
 <resources>
  <include location="Ui.qrc"/>
 </resources>
 <connections/>
</ui>
